ssh_check/datadog_checks/ssh_check/ssh_check.py

```diff
@@ -27,67 +26,37 @@ class CheckSSH(AgentCheck):
     SSH_SERVICE_CHECK_NAME = 'ssh.can_connect'
     SFTP_SERVICE_CHECK_NAME = 'sftp.can_connect'
 
-    OPTIONS = [
-        ('host', True, None, str),
-        ('port', False, 22, int),
-        ('username', True, None, str),
-        ('password', False, None, str),
-        ('private_key_file', False, None, str),
-        ('private_key_type', False, 'rsa', str),
-        ('sftp_check', False, True, bool),
-        ('add_missing_keys', False, False, bool),
-    ]
-
-    Config = namedtuple(
-        'Config',
-        [
-            'host',
-            'port',
-            'username',
-            'password',
-            'private_key_file',
-            'private_key_type',
-            'sftp_check',
-            'add_missing_keys',
-        ],
-    )
-
-    def _load_conf(self, instance):
-        params = []
-        for option, required, default, expected_type in self.OPTIONS:
-            value = instance.get(option)
-            if required and (not value or type(value) != expected_type):
-                raise Exception("Please specify a valid {0}".format(option))
-
-            if value is None or type(value) != expected_type:
-                self.log.debug("Bad or missing value for %s parameter. Using default", option)
-                value = default
-
-            params.append(value)
-        return self.Config._make(params)
```

Comment:
It seems that we are losing some config validation, but we can probably cover that later using config spec info.